sdk/servicebus/service-bus/src/core/linkEntity.ts

```diff
@@ -118,7 +118,7 @@ export class LinkEntity {
     await defaultLock.acquire(this._context.namespace.cbsSession.cbsLock, () => {
       return this._context.namespace.cbsSession.init();
     });
-    const tokenObject = await this._context.namespace.tokenProvider.getToken(this.audience);
+    const tokenObject = await this._context.namespace.tokenCredential.getToken(this.audience);
```

Comment:
`this._context.namespae.tokenCredential` can be either of type `SharedKeyCredential` or `TokenCredential`. We need to handle the two separately

Reply:
Oh, currently though we are only passing in SharedKeyCredential type right? 
I thought general update to work with all of newer identity handling was being covered in a different issue